com.unity.render-pipelines.high-definition/Runtime/Utilities/DeviceInfo.cs

```diff
@@ -0,0 +1,86 @@
+using System;
+using UnityEngine;
+#if UNITY_EDITOR
+using UnityEditor;
+#endif
+
+namespace UnityEngine.Rendering.HighDefinition
+{
+    public class DeviceInfo
+    {
+        public static int optimalThreadGroupSize;
+        public static int log2NumClusters; // / MSB of optimalThreadGroupSize, NumClusters is 1<<g_iLog2NumClusters
+        public static string kernelVariantSuffix;
+        public static bool preferComputeKernels;
+        public static bool requiresExplicitMSAAResolve;
+
+        static DeviceInfo()
+        {
+            ProbeDeviceInfo();
+        }
+
+        public static void ProbeDeviceInfo()
+        {
+            // Reset to defaults first
+            optimalThreadGroupSize = 64;
+            log2NumClusters = 6;
+            kernelVariantSuffix = "";
+            preferComputeKernels = true;
+            requiresExplicitMSAAResolve = true;
+
+            bool threadExecutionWidth32 = IsMobileBuildTarget;
+
+            if (SystemInfo.graphicsDeviceType == GraphicsDeviceType.Metal)
+            {
+                // TODO: Could conditionally enable threadExecutionWidth32/disable preferComputeKernels based on GPU
+                // SystemInfo.hasHiddenSurfaceRemovalOnGPU == true on Apple GPUs
+                requiresExplicitMSAAResolve = false;
```

Comment:
so this PR does not yet enable PLATFORM_LANE_COUNT_32 path when running on Mac/Apple GPU, but that's the ultimate goal for this, while cleaning up existing mobile/switch situation